src/agent/src/namespace.rs

```diff
@@ -117,3 +117,29 @@ pub fn setup_persistent_ns(logger: Logger, ns_type: &'static str) -> Result<Name
         path: new_ns_path.into_os_string().into_string().unwrap(),
     })
 }
+
+#[cfg(test)]
+mod tests {
+    use super::{setup_persistent_ns, NSTYPEIPC, NSTYPEUTS, PERSISTENT_NS_DIR};
+    use crate::{mount::remove_mounts, skip_if_not_root};
+
+    #[test]
+    fn test_setup_persistent_ns() {
+        skip_if_not_root!();
+
+        // TODO: Define if setup_persistent_ns really needs to own logger.
+        let logger = slog::Logger::root(slog::Discard, o!());
+        let res = setup_persistent_ns(logger, NSTYPEIPC);
+        assert!(res.is_ok());
+
+        let dir = vec![format!("{}/{}", PERSISTENT_NS_DIR, NSTYPEIPC)];
+        assert!(remove_mounts(&dir).is_ok());
```

Comment:
Thanks for raising this @ericho. However, I feel that the test should change the value of `PERSISTENT_NS_DIR` to a temporary value (using the `tempfile` create probably) since currently running this test would affect a running system.

Reply:
Sure, working on it. 

I review again the unit test from the Go version, I understand now that the temporary folders are used to override a global variable. This won't work the same way here. I need to find a way to either pass configuration to this function or maybe use some `#[cfg(test)]` blocks.